src/overlay/Overlay_D3D12.cpp

```diff
@@ -29,82 +28,186 @@ BOOL CALLBACK EnumWindowsProcMy(HWND hwnd, LPARAM lParam)
-            m_pd3dCommandList->Close() != S_OK)
-            return;
+    D3D12_DESCRIPTOR_HEAP_DESC rtvdesc = {};
+    rtvdesc.Type = D3D12_DESCRIPTOR_HEAP_TYPE_RTV;
+    rtvdesc.NumDescriptors = buffersCounts;
+    rtvdesc.Flags = D3D12_DESCRIPTOR_HEAP_FLAG_NONE;
+    rtvdesc.NodeMask = 1;
+    if (FAILED(m_pd3d12Device->CreateDescriptorHeap(&rtvdesc, IID_PPV_ARGS(&m_pd3dRtvDescHeap))))
+    {
+        spdlog::error("\tOverlay::InitializeD3D12() - failed to create RTV descriptor heap!");
+        return reset(this);
+    }
+
+    const SIZE_T rtvDescriptorSize = m_pd3d12Device->GetDescriptorHandleIncrementSize(D3D12_DESCRIPTOR_HEAP_TYPE_RTV);
+    D3D12_CPU_DESCRIPTOR_HANDLE rtvHandle = m_pd3dRtvDescHeap->GetCPUDescriptorHandleForHeapStart();
+    for (auto& context : m_frameContexts)
+    {
+        context.MainRenderTargetDescriptor = rtvHandle;
+        rtvHandle.ptr += rtvDescriptorSize;
+    }
 
-        for (UINT i = 0; i < buffersCounts; i++)
+    D3D12_DESCRIPTOR_HEAP_DESC srvdesc = {};
+    srvdesc.Type = D3D12_DESCRIPTOR_HEAP_TYPE_CBV_SRV_UAV;
+    srvdesc.NumDescriptors = 1;
+    srvdesc.Flags = D3D12_DESCRIPTOR_HEAP_FLAG_SHADER_VISIBLE;
+    if (FAILED(m_pd3d12Device->CreateDescriptorHeap(&srvdesc, IID_PPV_ARGS(&m_pd3dSrvDescHeap))))
+    {
+        spdlog::error("\tOverlay::InitializeD3D12() - failed to create SRV descriptor heap!");
+        return reset(this);
+    }
+    
+    for (auto& context : m_frameContexts)
+        if (FAILED(m_pd3d12Device->CreateCommandAllocator(D3D12_COMMAND_LIST_TYPE_DIRECT, IID_PPV_ARGS(&context.CommandAllocator))))
         {
-            pSwapChain->GetBuffer(i, IID_PPV_ARGS(&m_frameContexts[i].BackBuffer));
-            d3d12Device->CreateRenderTargetView(m_frameContexts[i].BackBuffer, nullptr, m_frameContexts[i].MainRenderTargetDescriptor);
+            spdlog::error("\tOverlay::InitializeD3D12() - failed to create command allocator!");
+            return reset(this);
         }
 
-        uintptr_t swapChainAddr = reinterpret_cast<uintptr_t>(pSwapChain);
-        m_pCommandQueue = *reinterpret_cast<ID3D12CommandQueue**>(swapChainAddr + kiero::getCommandQueueOffset());
+    if (FAILED(m_pd3d12Device->CreateCommandList(0, D3D12_COMMAND_LIST_TYPE_DIRECT, m_frameContexts[0].CommandAllocator, nullptr, IID_PPV_ARGS(&m_pd3dCommandList))) ||
+        FAILED(m_pd3dCommandList->Close()))
+    {
+        spdlog::error("\tOverlay::InitializeD3D12() - failed to create command list!");
+        return reset(this);
+    }
+
+    for (auto& context : m_frameContexts)
+        m_pd3d12Device->CreateRenderTargetView(context.BackBuffer, nullptr, context.MainRenderTargetDescriptor);
+
+    IMGUI_CHECKVERSION();
+    ImGui::CreateContext();
+    ImGuiIO& io = ImGui::GetIO();
+    ImGui::StyleColorsDark();
+    io.Fonts->AddFontDefault();
+    io.Fonts->GetTexDataAsRGBA32(&pixels, &width, &height);
```

Comment:
I don't really see the user of this get?

Reply:
Forgotten leftover from previous code I suppose. I believe I just copied this chunk in particular from former implementation. Will remove it ASAP.